### ISSUE TITLE
iterator_test.cpp fails

### ISSUE THREAD

**Issue by jmeintrup** [thm-mni-ii/sea#31]
```
iterator_test.cpp:44: FailureExpected equality of these values:
  count
    Which is: 1639
  setSize
    Which is: 5000
```

I have been getting that error from the iterator_test.
Did this test pass the travis build test?

**Comment by andrej-sajenko** [comment]
Yes the test passes on the travis build.

**Comment by andrej-sajenko** [comment]
Have you solved this issue?

**Comment by jmeintrup** [comment]
Yes, it seems to. It just worked after i tried it two more times.
The test uses some random generation, maybe that is the problem - even though it uses a fixed seed.

**Comment by shsr04** [comment]
Now also fails for me. #36

**Comment by dppl** [comment]
I'm going to look into it, however I'm not able to reproduce this behaviour locally.

> Now also fails for me. #36

Does it make a difference if you comment out line 28 in choicedictionaryiterator_test.cpp?

**Comment by shsr04** [comment]
> Does it make a difference if you comment out line 28 in choicedictionaryiterator_test.cpp?

Yes it does, but the test still fails. Now the actual value is `1624` instead of `3260`.

~~Let's see if Travis accepts it.~~

**Comment by dppl** [comment]
@andrej-sajenko Should we maybe disable the test for the choice dictionary iterator, so that unrelated pull requests get accepted?
I'm going to look further into it to find the problem, as it doesn't seem to be related to shuffling the test set.

@shsr04 Does the test also fail locally?

**Comment by shsr04** [comment]
Yes, it's the same as the Travis build.

**Comment by shsr04** [comment]
Maybe the `-O3` compiler flag introduced [one month ago](https://github.com/thm-mni-ii/sea/commit/f2f4bc26a8299cb3fd1e9fe749c5dba89ca11ac8#diff-af3b638bc2a3e6c650974192a53c7291) breaks the GCC built-in `__builtin_clzl` used in the iterator: https://github.com/thm-mni-ii/sea/blob/f4747b35eef2a292bbc22ebc6ebeb5cb58fe5b88/src/choicedictionaryiterator.cpp#L38
https://github.com/thm-mni-ii/sea/blob/f4747b35eef2a292bbc22ebc6ebeb5cb58fe5b88/src/choicedictionaryiterator.cpp#L61

**Comment by shsr04** [comment]
@dappels I have made several tries now. Whatever my approach for the CD iterator test is, it fails on Travis (but works locally)... Could this be a bug in the iterator itself? (some portability issue?)

**Comment by dppl** [comment]
I don't really know what causes the issue for you. As discussed previously it is not possible for me to reproduce that issue locally. 
Is there any sign of the iterator returning incorrect values when you run it locally? Be it in an implementation that relies on it or the unit test?

**Comment by shsr04** [comment]
The **Travis** build on #68 failed when I uncommented the test. (Locally, everything works.)
I also made a few different random tests (e.g. https://github.com/thm-mni-ii/sea/commit/641c791f43a56b1e6680873a2a67935a811d75ba, https://github.com/thm-mni-ii/sea/commit/e8ddd2e98b63d699597ac41215f6769ae44f31a3) to see if Travis accepts those. (Again, every one of those works locally.)

What's really interesting is that the experimental test https://github.com/thm-mni-ii/sea/commit/e8ddd2e98b63d699597ac41215f6769ae44f31a3 does not even use random number generation. The build status can be found here: https://travis-ci.org/thm-mni-ii/sea/builds/455994593

I don't know if this helps, but that's all the information I have.
